serverjs/analytics.js

```diff
@@ -1,3 +1,48 @@
+/* used in the now abandoned GetTokens attempt to get token counts from rules text. If  
+var Small = {
+  'zero': 0,
+  'one': 1,
+  'two': 2,
+  'three': 3,
+  'four': 4,
+  'five': 5,
+  'six': 6,
+  'seven': 7,
+  'eight': 8,
+  'nine': 9,
+  'ten': 10,
+  'eleven': 11,
+  'twelve': 12,
+  'thirteen': 13,
+  'fourteen': 14,
+  'fifteen': 15,
+  'sixteen': 16,
+  'seventeen': 17,
+  'eighteen': 18,
+  'nineteen': 19,
+  'twenty': 20,
+};
+
+var a, n, g;
+
+function text2num(a) {
+  return Small[a];  
+}
+*/
+function CheckContentsEqualityOfArray(target, candidate) {
```

Comment:
Can you put this in util.js instead and call it something like `arraySetEqual`?

I think this is a cleaner and more performant implementation:
```javascript
const candidateSet = new Set(candidate);
return target.length === candidate.length && target.every(x => candidateSet.has(x));
```